src/mappings/cellar.ts

```diff
@@ -0,0 +1,242 @@
+import { BigInt, log } from '@graphprotocol/graph-ts'
+
+import {
+  CellarContract,
+  AddedLiquidity,
+  RemovedLiquidity,
+  Rebalance,
+} from "../types/Cellar/CellarContract"
+import { Cellar, CellarInvest, Token } from "../types/schema"
+import {
+  initCellar,
+  loadCellar,
+  upsertNFLPs,
+  // TODO: REMOVE
+  // initUSDC,
+  // initWETH,
+} from '../utils/cellar'
+import { loadBundle } from '../utils/pricing'
+import { convertTokenToDecimal } from '../utils'
+
+// Hardcoded on the contract, used to figure out what the total collected fees were from the pool
+// TODO: should we fix the contract to emit the right value?
+
+let FEE_DENOMINATOR = BigInt.fromI32(10000)
+
+export function handleAddedLiquidity(event: AddedLiquidity): void {
+  log.debug('Invoked handleAddedLiquidity', [])
+  let cellarContract = CellarContract.bind(event.address)
+  let bundle = loadBundle()
+
+  let cellarAddress = event.address.toHex()
+  let cellar = Cellar.load(cellarAddress)
+  if (cellar == null) {
+    log.debug('Initialize Cellar: ', [cellarAddress])
+    cellar = initCellar(cellarContract, cellarAddress)
+  }
+
+  // odd that cellar type is Cellar | null here even though we init it above
+  upsertNFLPs(cellarContract, cellar as Cellar)
+
+  // TODO: DRY UP
+  let token0 = Token.load(cellar.token0)
+  let token1 = Token.load(cellar.token1)
```

Comment:
I think you need a check here for if either of these equal null, unless you want it to crash (which may be good in this case because it means the cellar entity is borked)